src/cobra/core/dictlist.py

```diff
@@ -150,35 +154,41 @@ def select_attribute(x):
         results._extend_nocheck(matches)
         return results
 
-    def _replace_on_id(self, new_object):
+    def _replace_on_id(self, new_object: Object) -> None:
         """Replace an object by another with the same id."""
         the_id = new_object.id
         the_index = self._dict[the_id]
         list.__setitem__(self, the_index, new_object)
 
     # overriding default list functions with new ones
-    def append(self, object):
-        """append object to end"""
-        the_id = object.id
+    def append(self, entity: Object) -> None:
+        """Append object to end."""
+        the_id = entity.id
         self._check(the_id)
         self._dict[the_id] = len(self)
-        list.append(self, object)
+        list.append(self, entity)
 
-    def union(self, iterable):
-        """adds elements with id's not already in the model"""
+    def union(self, iterable: Iterable) -> None:
```

Comment:
Is this too specific?

```suggestion
    def union(self, iterable: Iterable[Object]) -> None:
```